Restourant/Screens/SearchAddressBottomSheet/SearchAddressBottomSheet.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  SearchAddressBottomSheet.swift
+//  Restourant
+//
+//  Created by Иван Пономарев on 25.01.2023.
+//
+
+import Foundation
+import UIKit
+import SnapKit
+
+final class SearchAddressBottomSheet: UIViewController {
+    private let defaults = UserDefaults.standard
+    private var networkDataFetch: NetworkDataFetcherLogic?
+    private var addressArray = [DataClass]()
+
+
+    private let tableView: UITableView = {
+        let tableView = UITableView()
+        return tableView
+    }()
+
+    var clouse: ((String) -> ())?
+
+    lazy var searchBar:UISearchBar = UISearchBar()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+        setupSearchBar()
+        setupTableView()
+        searchBar.text = defaults.string(forKey: "cityKey")
+        postRequest(searchText: searchBar.text ?? "")
+    }
+
+    deinit{
+        print("deinit searchBottomSheet")
+    }
+
+    private func postRequest(searchText: String) {
+        networkDataFetch = NetworkDataFetcher()
+        networkDataFetch?.fetchAddres(searchTerm: searchText) { result in
+            guard let result = result else { return }
+            let arrayModels = result.suggestions.map({$0.data})
+            self.addressArray = arrayModels.filter{ i in i.street_with_type != nil}
```

Comment:
по названию  `i` непонятна суть хранящегося типа. я бы предложил так
```suggestion
            let suggestions = result.suggestions.map { $0.data }
            self.suggestions = suggestions.filter { suggestion in suggestions.street_with_type != nil }
```

и можно функцию фильтр записать еще так:
`filter { $0.street_with_type != nil }` но это на твое усмотрение